### PR TITLE
feat: playbook to deploy the snowdrop-bot to k8s

### PR DESCRIPTION
**Pull request by jacobdotcosta**
Optionally builds the application locally, and deploys it to k8s. It uses buildah in the process.

### CODE REVIEW

**Comment by cmoulliard** [comment]
- I think that we should also include, part of this role, the process to install/uninstall the K8S resources needed to deploy `snowdrop-bot` on K8s

- Do we have to install additional K8s resources such as : secret, service account, pvc, pc, ... ?
```
kc get deployment,service,ingress -n bot
NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
deployment.extensions/snowdrop-bot   1/1     1            1           84d

NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/snowdrop-bot   ClusterIP   10.107.48.179   <none>        8080/TCP   84d

NAME                              HOSTS   ADDRESS          PORTS   AGE
ingress.extensions/snowdrop-bot   *       195.201.87.126   80      84d
```
WDYT @jacobdotcosta

**Comment by jacobdotcosta** [comment]
No PV is being used currently, due to problems when starting the bot with the H2 database. I'll check the service and the ingress, how they have been deployed.

**Comment by jacobdotcosta** [comment]
I've created [snowdrop-bot issue 130](https://github.com/snowdrop-bot/snowdrop-bot/issues/130) to create a database and have some persistency.

File: snowdrop-bot/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
+# Introduction
+
+Ansible playbook to deploy the `snowdrop-bot` into k8s using [quay.io](https://quay.io/) registry.
+
+# Playbook
+
+This project consists of 2 playbooks:
+* [build](#build): builds the project
+* [deploy](#deploy): deploys the application to [quay.io](https://quay.io/) and restarts the k8s pod
```
Comment (cmoulliard) [inline snowdrop-bot/README.md:10]: We dont deploy the application but we push the image to the quay registry. Can you then rename `deploy` to `push` and do the same within the role/tasks ?

File: snowdrop-bot/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+
+# Introduction
+
+Ansible playbook to deploy the `snowdrop-bot` into k8s using [quay.io](https://quay.io/) registry.
```
Comment (cmoulliard) [inline snowdrop-bot/README.md:4]: We dont deploy the application but the role defined part of this PR is about compile or package/build/push the image to the quay registry. Can you then review this description section please ?

File: snowdrop-bot/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+| bot_version | yes |  | Version to be applied to the `snowdrop-bot` tag at [quay.io](https://quay.io/). |
+| build_folder | no | `/tmp/snowdrop-bot-deployment/{{ bot_version }}/` | Folder containing the built `snowdrop-bot`. |
+| skip_dw | no |  | When set to `true` the project won't be downloaded from [github](https://github.com/snowdrop-bot/snowdrop-bot) |
+
+## deploy
+
+This playbook deploys the `snowdrop-bot` by performing the following tasks:
+1. Create local image
+1. Push image to [quay.io](https://quay.io/) with 2 tags, one for the version and another for the `latest`
+1. Scale down and up again the pod (the pod must be deployed against the `latest` version)
```
Comment (cmoulliard) [inline snowdrop-bot/README.md:31]: I propose to move the last step described here into another section "Rollout" as what you describe concerns the process to push an image to quay registry and that you use instead the following command to 

```
kubectl rollout restart -h
Restart a resource.

     Resource will be rollout restarted.

Examples:

  # Restart a deployment
  kubectl rollout restart deployment/nginx
```

See: https://github.com/kubernetes/kubernetes/issues/33664#issuecomment-633658979

File: snowdrop-bot/ansible/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        - "latest"
+
+    - name: "Scale down pod"
+      shell: "{{ kube_cli_tool }} scale --replicas=0 deployment snowdrop-bot -n bot {{ kube_cfg_append }}"
+
+    - name: "Wait 10 seconds"
+      pause:
+        seconds: 10
+
+    - name: "Scale up pod"
```
Comment (cmoulliard) [inline snowdrop-bot/ansible/deploy.yml:56]: To be moved into another file `rollout.yml`

Comment (cmoulliard) [reply snowdrop-bot/ansible/deploy.yml:56]: To be renamed as the title to `buikld-push.yml`

**Comment by cmoulliard** [comment]
PR is still missing what I commented 13days ago ;-)

**Comment by jacobdotcosta** [comment]
> PR is still missing what I commented 13days ago ;-)

I have reviewed the descriptors and this information is provided from the application itself in `target/kubernetes` folder. The `kubernetes.yml` file generated by quarkus includes the deployment, the service and the ingress so IMO these files don't fit here in this playbook. 

Since we don't have a testing environment I'd prefer leaving this playbook as is for now as it works to update the bot as create a new issue to implement this as a future improvement.

WDYT @cmoulliard ?

File: snowdrop-bot/ansible/deploy.yml (hunk 4 above)
Comment (jacobdotcosta) [reply snowdrop-bot/ansible/deploy.yml:56]: Changed the rollout tasks to a new playbook.
What do you mean with _To be renamed as the title to `buikld-push.yml`_?

**Comment by cmoulliard** [comment]
> I have reviewed the descriptors and this information is provided from the application itself in `target/kubernetes` folder. The `kubernetes.yml` file generated by quarkus includes the deployment, the service and the ingress so IMO these files don't fit here in this playbook.

Ok. LGTM then

**Comment by cmoulliard** [comment]
Go ahead and merge it then

**Pull request merged**